File: 0974-subarray-sums-divisible-by-k/0974-subarray-sums-divisible-by-k.py
class Solution:
    def subarraysDivByK(self, nums: List[int], k: int) -> int:
        count=0
        Sum=0
        check=dict()
        check[0]=1
        for num in nums:
            Sum+=num
            rem=Sum%k
            if rem<0:
                rem+=k
            if rem in check:
                count += check[rem]
                check[rem] +=1
            else:
                check[rem]=1
        return count

File: 0974-subarray-sums-divisible-by-k/0974-subarray-sums-divisible-by-k.java
class Solution {
    public int subarraysDivByK(int[] nums, int k) {
        HashMap<Integer,Integer> h=new HashMap<>();
        h.put(0,1);
        int total=0,ans=0;
        for(int n:nums){
            total+=n;
            int rem=total%k;
            if(rem<0){
                rem+=k;
            }
            if(h.containsKey(rem)){
                ans+=h.get(rem);
                h.put(rem,h.get(rem)+1);
            }else{
                h.put(rem,1);
            }
        }
        return ans;
    }
}